src/librustc_errors/diagnostic_builder.rs

```diff
@@ -385,3 +385,22 @@ impl<'a> Drop for DiagnosticBuilder<'a> {
         }
     }
 }
+
+#[macro_export]
+macro_rules! struct_span_err {
+    ($session:expr, $span:expr, $code:ident, $($message:tt)*) => ({
+        $session.struct_span_err_with_code(
+            $span,
+            &format!($($message)*),
+            $crate::error_code!($code),
+        )
+    })
+}
+
+#[macro_export]
+macro_rules! error_code {
+    ($code:ident) => {{
+        let _ = $code;
```

Comment:
It's one of my goals to remove this (mainly for incremental benefits and possibly internationalization as a stretch goal), but I wanted to leave it for a follow-up since there might be objections. As I understand it, we want to both catch unused error codes (I believe tidy does this) and used unregistered error codes (tidy doesn't catch this).

Reply:
Ah, if tidy still makes the check, then let's leave it.